### PR TITLE
Skeleton documentation

### PR DESCRIPTION
**Pull request by alexpearce**
Using [Sphinx](http://www.sphinx-doc.org) to generate a website containing documentation.

A preview of the HTML output is available here: http://apearce.web.cern.ch/apearce/scikit-hep/.

The structure is similar to that proposed by @eduardo-rodrigues. The content is almost non-existant 😄  This PR can focus on what we want to present. There's also the question of what style we want, but we should try to keep this PR focussed, lest the thread get hundreds of comments long.

### CODE REVIEW

**Comment by eduardo-rodrigues** [comment]
Hi Alex, again, many thanks for this first version of the website!
By construction I'm already quite happy with the bulk given our discussions prior to this MR. So others may have more comments. In any case I put here some of my feedback for (re)consideration (not comprehensive but, as Alex says, we need to focus):
1) To me the structure if largely find and ready to go as first version of the site.
2) Main structure change: I reckon it will be better to have the About, ... Affiliated packages items of the navigation menu as a header on all site pages, with a background colour. This is just about the first thing to decide.
3) I would also have a little footer. For now it would just say "(c) 2016-2017, Scikit-HEP Developers.".
4) I am fine with the basic idea of the logo but do not find the colours suitable. I have Alex an idea and should send an email about that. Basically the purple contours would become light grey and the words in a soft blue that would match the background colour for the header and the footer. To me that's nice and does not harm the eyes. The present colours are basically impossible to match with any colours for the site. I would also suggest that we have a version of the logo with "Scikit-HEP" written on a line, to be used for headers/footers in présentations, for example.
5) I would prefer to rename doc/ -> docs/.
Last but not least: let's please converge on a PR we can accept so that we still have time to fill in some info before the public presentation on the 27th ;-). I certainly have a few ideas on what to add to these pages Alex prepared.
I enumerated my comments to make it easy for everyone to comment further.

**Comment by eduardo-rodrigues** [comment]
Alex, a trivial detail I forgot to ask about: I do not understand why you needed to move the banner to a script. The banner was put in __init__.py because there are high chances that it will be used in various bits of code. So a script is not the place for it. It must be rather easy to fix the issue you found in some other way, right?

**Comment by eduardo-rodrigues** [comment]
Alex, here is the feedback from our team meeting:
1) Consensus with the present layout of the site.
2) Overall the site looks very good too on mobiles. Suggestion to test the layout on various mobiles/tablettes/... with the site http://mobiletest.me/.
3) We converged on the colour scheme for the logo, so gray and blue as sent via email. An updated logo will be made available in the next few days.
4) Issue that one gets a lot of white space to the left of the left vertical menu when expanding the browser horizontally. Probably a little css feature to fine-tune/
5) To mark a difference one could use one of the colour of the logo for the left navigation menu.
Bottom line: thanks very much for the work and we are almost there!

**Comment by eduardo-rodrigues** [comment]
BTW, can you update your private branch with the head of the project and update this PR? Just so that we avoid inconsistencies. Thanks.

**Comment by alexpearce** [comment]
I've rebased this branch onto master. You'll have to force pull the branch if you want to make more changes @eduardo-rodrigues.

**Comment by eduardo-rodrigues** [comment]
So far I made the changes directly on GitHub, I mean the updates to your PR. I will wait for the updates following the feedback.

**Comment by alexpearce** [comment]
Some feedback at last 😄 

> I do not understand why you needed to move the banner to a script.

Because it was in the docstring for the module, and so was parsed as invalid reStructed text. I moved it to the script because I didn't see it being used elsewhere.

> I would prefer to rename doc/ -> docs/.

I went with `doc/` as it's the conventional name amongst Python projects. It can be changed if you still prefer `docs/`.

---

I agree that the layout needs continuous testing on several devices. The default Sphinx theme, that we're using, has already been tested to some extent. Any modifications we make should be checked.

I've updated the hosted version: http://apearce.web.cern.ch/apearce/scikit-hep/

**Comment by eduardo-rodrigues** [comment]
I will accept this PR now. There are still things to sort out such as more info and background colour, etc. But that can come with updates. The most important thing right now is to have the website live.
